### PR TITLE
kiln: Desk publisher switcher (UIP-0111)

### PR DESCRIPTION
**Pull request by tinnus-napbus**
Allows app publishers to send a poke to all those subscribed for desk updates requesting they change the source of updates to a new ship/desk. Such requests are recorded in kiln's state, and the recipients can then approve or reject them. Upon approval, all existing syncs from the the old ship/desk will be changed to sync from the new ship/desk. This allows app publishers to more easily migrate the distribution ship.

see [UIP-0111](https://github.com/urbit/UIPs/blob/main/UIPS/UIP-0111.md) for more details

### CODE REVIEW

File: pkg/arvo/mar/kiln/jump-ask.hoon [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+/?    310
+|_  req=[old=dock new=dock]
+::
+++  grow
+  |%
+  ++  noun  req
+  --
+++  grab
+  |%
+  ++  noun  ,[dock dock]
```
Comment (liam-fitzgerald) [inline pkg/arvo/mar/kiln/jump-ask.hoon:13]: prefer `+$` to explicitly invoking spec mode

File: pkg/arvo/mar/kiln/jump.hoon [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/-  h=hood
+|_  jum=jump:h
+::
+++  grow
+  |%
+  ++  noun  jum
+  ++  json
+    =,  enjs:format
+    |^  ^-  ^json
+    ?-    -.jum
```
Comment (liam-fitzgerald) [inline pkg/arvo/mar/kiln/jump.hoon:13]: minor nit:
the `%+  frond  'foo'` can be moved above the wutlus for brevity, i.e.
```hoon
%+  frond  -.jum
```

File: pkg/arvo/lib/hood/kiln.hoon [hunk 3]
```diff
@@ -731,6 +758,80 @@
...
+  =.  hop  (~(put by hop) old new)
+  abet:(emit %give %fact ~[/jumps] %kiln-jump !>([%add old new]))
+::
+++  poke-jump-opt
+  |=  [old=dock new=dock yea=?]
+  ?~  got=(~(get by hop) old)
+    ~>  %slog.(fmt "no jump request for {<p.old>}/{(trip q.old)}")
+    abet
+  ?.  =(new u.got)
+    =/  txt-old  "{<p.old>}/{(trip q.old)}"
```
Comment (liam-fitzgerald) [inline pkg/arvo/lib/hood/kiln.hoon:803]: I'd prefer `{(scow %p p.old)}` but maybe that's just me

**Review by liam-fitzgerald — APPROVED** [review]
like, minor nits, but LGTM

**Review by belisarius222 — APPROVED** [review]

**Comment by belisarius222** [comment]
This is some pretty Hoon, @tinnus-napbus.

**Pull request merged**